### ISSUE TITLE
Willow fails with Photshop 3.0 jpegs

### ISSUE THREAD

**Issue by mope** [torchbox/Willow#12]
Attempting to load a jpeg with the following header:

```
ÿØÿí,Photoshop 3.08BIMí,,ÿá¸http://ns.adobe.com/xap/1.0/<?xpacket begin='ï»¿' id='W5M0MpCehiHzreSzNTczkc9d'?><x:xmpmeta xmlns:x='adobe:ns:meta/' x:xmptk='XMP toolkit 3.0-29, framework 1.6'>
```

Causes Willow to fail with:

```
Traceback (most recent call last):
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/views/decorators/cache.py", line 38, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/usr/local/django/virtualenvs/gwwagtail/src/wagtail-master/wagtail/wagtailimages/views/chooser.py", line 77, in chooser
    'will_select_format': request.GET.get('select_format')
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/shortcuts.py", line 50, in render
    return HttpResponse(loader.render_to_string(*args, **kwargs),
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/loader.py", line 178, in render_to_string
    return t.render(context_instance)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 148, in render
    return self._render(context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 142, in _render
    return self.nodelist.render(context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 844, in render
    bit = self.render_node(node, context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 858, in render_node
    return node.render(context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 844, in render
    bit = self.render_node(node, context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/base.py", line 858, in render_node
    return node.render(context)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/django/template/defaulttags.py", line 208, in render
    nodelist.append(node.render(context))
  File "/usr/local/django/virtualenvs/gwwagtail/src/wagtail-master/wagtail/wagtailimages/templatetags/wagtailimages_tags.py", line 55, in render
    rendition = image.get_rendition(self.filter)
  File "./images/models.py", line 167, in get_rendition
    generated_image, output_format = filter.run(self, BytesIO())
  File "/usr/local/django/virtualenvs/gwwagtail/src/wagtail-master/wagtail/wagtailimages/models.py", line 318, in run
    with image.get_willow_image() as willow:
  File "/opt/tbx/python/3.4/lib/python3.4/contextlib.py", line 59, in __enter__
    return next(self.gen)
  File "./images/models.py", line 82, in get_willow_image
    yield WillowImage.open(image_file)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/willow/image.py", line 62, in open
    initial_backend = cls.find_loader(image_format)
  File "/usr/local/django/virtualenvs/gwwagtail/lib/python3.4/site-packages/willow/image.py", line 143, in find_loader
    raise cls.LoaderError("Cannot find backend that can load this image file")
willow.image.LoaderError: Cannot find backend that can load this image file
```

**Comment by zerolab** [comment]
@mope can you add a sample image as well?

**Comment by mope** [comment]
@zerolab with pleasure:

![comic_only_english](https://cloud.githubusercontent.com/assets/1159140/7270602/40bc695e-e8d5-11e4-884e-44bd2f92cdc2.jpg)

**Comment by asmaps** [comment]
I seem to have the same problem. I uploaded a JPG into my wagtail and now I can't even access the "Images" overview page. It causes a 500 with the error mentioned above:
```
LoaderError
 Cannot find backend that can load this image file
```

The problematic point I extracted from the traceback would be the call of `imghdr.what(f)` here: https://github.com/torchbox/Willow/blob/master/willow/image.py#L55

This seems to return `None` which triggers the exception here:
https://github.com/torchbox/Willow/blob/master/willow/image.py#L140

I am using python 3.4 I can provide a full traceback if needed. Probably it's a bug in python imghdr, but I think until it is fixed there there should be some workaround in willow to prevent crashes.  
I will try to make a pull request, but I don't really have a clue now on how to fix it in a good way. Maybe fall back to file extension (*.jpg).
